File: typescript/createPool/create.raydium.pool.test.ts
import { Keypair, VersionedTransaction } from "@solana/web3.js";
import { bs58 } from "@project-serum/anchor/dist/cjs/utils/bytes";
import { configDotenv } from "dotenv";
configDotenv();

const pk = process.env.PRIVATE_KEY;
const test = async () => {
  try {
    const private_key = pk || "";
    const wallet = Keypair.fromSecretKey(bs58.decode(private_key));
    console.log("wallet:", wallet.publicKey.toBase58());

    const param = {
      market_id: "",
      wallet_address: wallet.publicKey.toBase58(),
      mint: "",
      tokenAmount: 1000000,
      solAmount: 0.1,
      // startTime: Date.now() + 2 * 60 * 60, //2hrs optional parameter
      tip: 0.00001,
    };

    const url = "https://api.solanaportal.io/api/create/pool";
    const response = await fetch(url, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(param),
    });

    if (response.status === 200) {
      // successfully generated transaction
      const data = await response.json();
      const txnBuffer = Buffer.from(data, "base64");
      const txn = VersionedTransaction.deserialize(txnBuffer);
      txn.sign([wallet]);
      const signedTxnBuffer = bs58.encode(txn.serialize());
      const jitoResponse = await fetch(
        `https://tokyo.mainnet.block-engine.jito.wtf/api/v1/transactions`,
        {
          method: "POST",
          headers: {
            "Content-Type": "application/json",
          },
          body: JSON.stringify({
            jsonrpc: "2.0",
            id: 1,
            method: "sendTransaction",
            params: [signedTxnBuffer],
          }),
        }
      );
      if (jitoResponse.status === 200) {
        const signature = (await jitoResponse.json()).result;
        console.log("- txn succeed", "https://solscan.io/tx/" + signature);
      } else {
        console.log("- txn failed, please check the parameters", data);
      }
    } else {
      console.log(response.statusText); // log error
    }
  } catch (e: any) {
    console.error(e.message);
  }
};

test();

File: python/createPool/create.raydium.pool.test.py
#!/usr/bin/env python3
from dotenv import load_dotenv
import os
import base64
import base58
import requests
from solana.keypair import Keypair
from solana.transaction import VersionedTransaction

def main():
    # Load PRIVATE_KEY from .env
    load_dotenv()
    private_key_b58 = os.getenv("PRIVATE_KEY")
    if not private_key_b58:
        raise EnvironmentError("Missing PRIVATE_KEY in environment")

    # Build your Solana wallet Keypair
    secret = base58.b58decode(private_key_b58)
    wallet = Keypair.from_secret_key(secret)
    print(f"wallet: {wallet.public_key}")

    # Prepare the create-pool payload
    param = {
        "market_id":    "",               # your pool market ID
        "wallet_address": str(wallet.public_key),
        "mint":          "",              # token mint address
        "tokenAmount":  1_000_000,        # amount of tokens to deposit
        "solAmount":    0.1,              # amount of SOL to deposit
        # Optional: start time in milliseconds since epoch
        # "startTime":    int((time.time() + 2*60*60) * 1000),
        "tip":          0.00001,          # priority fee
    }

    # 1) Request the unsigned pool-creation txn
    url = "https://api.solanaportal.io/api/create/pool"
    resp = requests.post(url, json=param)
    if resp.status_code != 200:
        print("Error generating pool-creation txn:", resp.text)
        return

    # 2) Deserialize & sign locally
    txn_b64   = resp.json()  # base64-encoded VersionedTransaction
    txn_bytes = base64.b64decode(txn_b64)
    txn       = VersionedTransaction.deserialize(txn_bytes)
    txn.sign([wallet])

    # 3) Re-serialize and base58-encode
    signed_b58 = base58.b58encode(txn.serialize()).decode()

    # 4) Send it via Jito RPC
    jito_url = "https://tokyo.mainnet.block-engine.jito.wtf/api/v1/transactions"
    jito_payload = {
        "jsonrpc": "2.0",
        "id":      1,
        "method":  "sendTransaction",
        "params":  [signed_b58],
    }
    jito_resp = requests.post(jito_url, json=jito_payload)
    if jito_resp.status_code == 200:
        signature = jito_resp.json().get("result")
        print("- txn succeed:", f"https://solscan.io/tx/{signature}")
    else:
        print("- txn failed, please check the parameters:", jito_resp.text)

if __name__ == "__main__":
    main()
